src/kw_config_loader.sh

```diff
@@ -142,7 +142,14 @@ function parse_configuration()
 function load_configuration()
 {
   parse_configuration "$KW_ETC_DIR/$CONFIG_FILENAME"
-  parse_configuration "$HOME/.kw/$CONFIG_FILENAME"
+
+  # XDG_CONFIG_DIRS is in order of preference, but we must first read
+  # from the folders with less preference
+  while read -d ':' -r dir; do
+    parse_configuration "$(rev <<< "$dir")/$KWORKFLOW/$CONFIG_FILENAME"
+  done <<< "$(rev <<< "${XDG_CONFIG_DIRS:-"/etc/xdg"}"):"
```

Comment:
What is this XDG order preference? Do we really need it? Can we make this easier to read? Also, can you expand the explanation about the order in the comment?

Reply:
We need it to be compliant with the specification... I made some changes for the ease of reading and added comments.